block/p2p.go

```diff
@@ -0,0 +1,61 @@
+package block
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/dymensionxyz/dymint/p2p"
+	"github.com/dymensionxyz/dymint/types"
+	"github.com/tendermint/tendermint/libs/pubsub"
+)
+
+// onReceivedBlock receives a block received event from P2P, saves the block to a cache and tries to apply the blocks from the cache.
+func (m *Manager) onReceivedBlock(event pubsub.Message) {
+	eventData, ok := event.Data().(p2p.P2PBlockEvent)
+	if !ok {
+		m.logger.Error("onReceivedBlock", "err", "wrong event data received")
+		return
+	}
+	var source blockSource
+	switch event.Events()[p2p.EventTypeKey][0] {
+	case p2p.EventNewBlockSyncBlock:
+		source = blocksyncBlock
+	case p2p.EventNewGossipedBlock:
+		source = gossipedBlock
+	default:
+		m.logger.Error("onReceivedBlock", "err", "wrong event type received")
```

Comment:
can we add here the event type received?

Reply:
done